### DISCUSSION TITLE
A use case I was interested in: getting the HTML string with a GET request to a URL, then manipulating the String to get Open Graph data

### DISCUSSION THREAD

**Discussion by pbiggar** [darklang/dark#4350] [Feedback]
A user says:
>Overall, some feedback so far I might have would be (might mostly be due to my incompetence, I’d consider myself only an amateur developer so far):
>
>Requests I care more about
>- Looking forward to even more documentation / examples! Particularly related to the built-in types
>    - a use case I was interested in: getting the HTML string with a GET request to a URL, then manipulating the String to get Open Graph data)
>    - Another area where I was looking for more info is how to use Blocks, for instance with DB::query. A complex example of getting something from a DB, parsing that, then using that info to query for something from another DB would be helpful. 
>- Wondering if a standardized auth setup + client-side SDK is on the roadmap in the future? 
>
>Requests that are less imminent
>- It was sometimes unexpectedly awkward if I wanted to insert an if/then statement in the middle of existing code to rework my logic. It might be my lack of familiarity with the language; I’ve mostly only worked with imperative languages before. I would expect it would be very enlightening if ya’ll could document some common antipatterns of working with the Dark language.
>- Wish there was support for zoom. It would be a 10x improvement especially if support for better structuring of the canvas isn’t available yet—zoom would make that totally fine
>    - Maybe some type of “label” you can put on the canvas would be helpful. When using Figma I sometimes put a random Text box somewhere to mark areas (maybe an antipattern, not sure)
>- a more out of the way bug, but if I drag an element on the canvas (like a handler), and also pan with my scroll, the item doesnt follow my cursor
>
>Things I like:
>- May take awhile to get used to, but I love how Tab navigation jumps right to what I need to work on
>- obviously the breakthrough of instant deploy
>
>I’ve been showing tons of my peers Dark and saying its the best thing since sliced bread. Setting up infra was my biggest block when I was first learning to create something and put it out there in the world. Showing off a hello world example is really refreshing.

**Comment by StachuDotNet** [comment]
> Wondering if a standardized auth setup + client-side SDK is on the roadmap in the future?

I wonder what this means?

**Reply by pbiggar** [reply]
I understood this as a built-in user type (with auth including social login), plus a JS library that can interact with it. Firebase provides one for example.
